CoreTableView/test/unit/src/au/gov/asd/tac/constellation/views/tableview2/io/TableViewPreferencesIOUtilitiesNGTest.java

```diff
@@ -0,0 +1,214 @@
+    
+    @Test
+    public void getPreferencesSinglePreference() throws IOException {
+        final ObjectMapper objectMapper = new ObjectMapper();
+        final JsonNode jsonNode = objectMapper.readTree(
+                new FileInputStream(getClass().getResource("resources/transaction-preferences.json").getPath())
+        );
+        
+        jsonIOStaticMock.when(() -> JsonIO.loadJsonPreferences("TableViewPreferences", "transaction-"))
+                .thenReturn(jsonNode);
+        
+        final ThreeTuple<List<String>, Tuple<String, TableColumn.SortType>, Integer> preferences
+                = TableViewPreferencesIOUtilities.getPreferences(GraphElementType.TRANSACTION, 2);
+        
+        final ThreeTuple<List<String>, Tuple<String, TableColumn.SortType>, Integer> expected = ThreeTuple.create(
+                List.of("ABC", "DEF"),
+                Tuple.create("DEF", TableColumn.SortType.ASCENDING),
+                5
+        );
+        
+        assertEquals(expected, preferences);
+    }
+    
+    @Test
+    public void getPreferencesNullPrefs() throws IOException {       
+        jsonIOStaticMock.when(() -> JsonIO.loadJsonPreferences("TableViewPreferences", "vertex-"))
+                .thenReturn(null);
+        
+        final ThreeTuple<List<String>, Tuple<String, TableColumn.SortType>, Integer> preferences
+                = TableViewPreferencesIOUtilities.getPreferences(GraphElementType.VERTEX, 2);
+        
+        final ThreeTuple<List<String>, Tuple<String, TableColumn.SortType>, Integer> expected = ThreeTuple.create(
+                Collections.emptyList(),
+                Tuple.create("", TableColumn.SortType.ASCENDING),
+                500
+        );
+        
+        assertEquals(expected, preferences);
+    }
+    
+    @Test
+    public void savePreferences() throws IOException {
+        // TODO Find a better solution for this. Because of this limitation these tests
+        //      will not be run on the CI server.
+        if (!GraphicsEnvironment.isHeadless()) {
+            // Interestingly once you throw the skip exception it doesn't call the tear down class
+            // so we need to instantiate the static mocks only once we know we will be running the
+            // tests.
+            new JFXPanel();
+        
+            nbPreferencesStaticMock.when(() -> NbPreferences.forModule(ApplicationPreferenceKeys.class))
+                    .thenReturn(null);
+            applicationPrefKeysStaticMock.when(() -> ApplicationPreferenceKeys.getUserDir(null))
+                    .thenReturn(System.getProperty("java.io.tmpdir"));
+
+            final ObservableList<TableColumn<ObservableList<String>, ? extends Object>> columns = mock(ObservableList.class);
+            when(columns.size()).thenReturn(4);
+
+            final ObservableList<TableColumn<ObservableList<String>, ? extends Object>> sortOrder = mock(ObservableList.class);
+            when(sortOrder.isEmpty()).thenReturn(Boolean.FALSE);
```

Comment:
can return the primitive `false` here rather than the object. Similar for the ones below this

Reply:
changed to primative.